sites/public/pages/applications/preferences/all.tsx

```diff
@@ -0,0 +1,311 @@
+import React, { useCallback, useMemo, useState, useEffect } from "react"
+import { useForm } from "react-hook-form"
+import {
+  AlertBox,
+  Form,
+  FormCard,
+  ProgressNav,
+  Field,
+  t,
+  ExtraField,
+  PREFERENCES_FORM_PATH,
+  PREFERENCES_NONE_FORM_PATH,
+  ExpandableContent,
+  Button,
+  AppearanceStyleType,
+  resolveObject,
+  mapPreferencesToApi,
+  mapApiToPreferencesForm,
+} from "@bloom-housing/ui-components"
+import FormsLayout from "../../../layouts/forms"
+import FormBackLink from "../../../src/forms/applications/FormBackLink"
+import { useFormConductor } from "../../../lib/hooks"
+import { FormMetadataOptions } from "@bloom-housing/backend-core/types"
+
+const PreferencesAll = () => {
+  const { conductor, application, listing } = useFormConductor("preferencesAll")
+  const preferences = listing?.preferences
+
+  const currentPageSection = 4
+
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  const { register, setValue, watch, handleSubmit, errors, getValues, trigger } = useForm({
+    defaultValues: {
+      application: { preferences: mapApiToPreferencesForm(application.preferences) },
+    },
+  })
+
+  /*
+    It creates string path for each preference option - for error validation purposes
+  */
+  const preferenceOptionObjectPaths = useMemo(() => {
+    return preferences?.reduce((acc, item) => {
```

Comment:
If I understood correctly you are mapping props to acc object to it seems easier to use `map` operator instead, does that make sense?

Reply:
This function produces more elements than preferences length, so it's not possible to use `map`.
I created this, for validation purposes (it's used in e.g 262 line, in the `somethingIsChecked` validator) - react-hook-form is not ideal in some custom situations :D

Reply:
I must admit I find this portion of the code hard to follow.￼ Maybe take one more look at it, and at the very least add some comments about what's happening?￼